### PR TITLE
Add support for parsing Link Header with variable whitespace

### LINKED ISSUE #5667: Link Header pagination fails when there's no space between ';' and 'rel'
If links in a Link header do not contain a space between `;` and `rel` then invoke-webrequest fails to populate the RelationLink property 
```
<http://localhost:8080/PowerShell?test=linkheader&maxlinks=5&linknumber=1>; rel="self"
<http://localhost:8080/PowerShell?test=linkheader&maxlinks=5&linknumber=1>;rel="self"
```
Not having a space is valid (more appropriate, even) per the specification: https://tools.ietf.org/html/rfc5988 page 6

### For reference 
  #5265 [Feature] Add Multiple Link Header Support 
#### Code
https://github.com/PowerShell/PowerShell/blob/9e99d596212bc6e2b92af6700889f7698492474d/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs#L835
#### Test
https://github.com/PowerShell/PowerShell/blob/9e99d596212bc6e2b92af6700889f7698492474d/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1#L797

I think the appropriate fix would be to change `\\s` to `\\s*` to allow for 0 or more whitespace rather than requiring a single whitespace 

- Current regex not working: https://regex101.com/r/QbglHf/1
- Proposed modification: https://regex101.com/r/QbglHf/2

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
add support for variable whitespace in link header

## PR Summary

Thanks to the @cruscio's analysis, it appears that the Link Header allows for any amount of whitespace or even no whitespace.  The currently explicitly expect a single whitespace after the semicolon in the Link Header before the `Rel` property.

Fix is to change the regex to allow for variable or no whitespace and associated tests to handle these two cases.  Changed `GetLink` helper to support passing in whitespace to use.

Fix https://github.com/PowerShell/PowerShell/issues/5667

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -1789,7 +1789,7 @@ internal void ParseLinkHeader(HttpResponseMessage response, System.Uri requestUr
 
             // we only support the URL in angle brackets and `rel`, other attributes are ignored
             // user can still parse it themselves via the Headers property
-            string pattern = "<(?<url>.*?)>;\\srel=\"(?<rel>.*?)\"";
+            string pattern = "<(?<url>.*?)>;\\s*rel=\"(?<rel>.*?)\"";
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: Please do not introduce any overlapping terms

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs (hunk 1 above)
Comment (TravisEz13) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: Resolved, I misread the Regex

**Review by TravisEz13 — APPROVED** [review]

**Review by markekraus — DISMISSED** [review]
LGTM

**Review by markekraus — CHANGES_REQUESTED** [review]
LGTM, but need to update WebListener documentation https://github.com/PowerShell/PowerShell/tree/master/test/tools/WebListener#link

**Comment by SteveL-MSFT** [comment]
@markekraus added documentation!

**Review by markekraus — APPROVED** [review]

**Pull request merged**